README.md

```diff
@@ -166,13 +166,103 @@ Similarly launch the perception_node.py
 
+## Namespaced Simulations
 
+By defining each simulation under its own namespace, we eliminate the need to generate separate YAML files for each experiment. This ensures that each simulation runs independently, even though the parameters and results are stored in shared directories. The use of ROS2 launch files enables seamless namespace management for multiple simulations.
 
+To set up the parallel simulation:
 
+1 .Generate YAML and Launch Files for Each Namespace:
+ 
+ The script_generator.py automatically generates the required YAML files for each parameter combination and creates corresponding launch files under different namespaces.
+
+```
+ros2 run virtual_shake_robot_pybullet script_generator.py
+```
+This script will:
+
+- Generate YAML files with combinations of physics parameters for each experiment.
+- Create corresponding launch files for each generated YAML file.
+- Additionally, create a master launch file (e.g., {mesh_file_name}_master_launch.py) that launches all simulations in parallel.`
+
+2. Run the Master Launch File:
+
+```
+ros2 launch virtual_shake_robot_pybullet {mesh_file_name}_master_launch.py
+```
+
+3. Monitor the results
+
+    Each simulation will log its results in separate files, organized by namespace. This allows you to later compare the behaviors of PBRs across different parameter combinations.
+
+
+
+# Running Parallel Simulations on a Server (Manual Setup)
+
+If you need to run the parallel simulations on a server with more computational power, you can manually set up the environment using a base ROS2 Docker image or build your own Dockerfile. Here's a step-by-step guide:
+
+## 1. Using a Pre-Built ROS2 Docker Image
+
+Start by pulling a base ROS2 Docker image that includes the ROS2 Humble release:
+```bash
+docker pull osrf/ros:humble-desktop
+```
+Manual Setup:
+
+Once you have the image, you will need to install the following dependencies manually:
+
+    pip (for Python package management)
+    pybullet
+    Any other dependencies required by your project.
+
+
+start the docker container :
+
+```
+docker run -it --name vsr_ros2_simulation osrf/ros:humble-desktop
+```
+
+This will start the container with the dependencies.
+
+So now follow the README.md for the vsr setup
+
+
+Apptainer (formerly Singularity): Apptainer is more suited for shared server environments where you might not have root or sudo access, as it runs containers in a user space.
+
+- Convert the docker image to an Apptainer Image
+
+```
+apptainer build vsr_simulation.sif docker://vsr_simulation
+
+```
+ - Run the container
+
+```
+apptainer run vsr_simulation.sif
+```
+
+Follow Similar steps as the docker container
+
+# Memory Leakage Issue
+
+During the Virtual Shake Robot (VSR) simulations, a memory leakage issue was observed, especially in long-running or parallel experiments. This issue was mainly due to continuous data logging processes, where trajectory data was recorded over time, leading to increased memory usage.
```

Comment:
This is not true. The memory leak issues are caused by several factors: 1) data recording, 2) creating and deleting plots in the callback function, 3) loading and deleting a URDF model.